test/image/mocks/multicategory-inside-ticks.json

```diff
@@ -0,0 +1,76 @@
+{
+  "data": [
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3" ]
+      ],
+      "y": [1, 2, 3, 1, 3, 2, 3, 1]
+    },
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3"]
+      ],
+      "y": [1.12, 2.15, 3.07, 1.48, 2.78, 1.95, 2.54, 0.64]
+    },
+
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3"]
+      ],
+      "y": [1.12, 2.15, 3.07, 1.48, 2.78, 1.95, 2.54, 0.64],
+      "xaxis": "x2",
+      "yaxis": "y2"
+    },
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3" ]
+      ],
+      "y": [1, 2, 3, 1, 3, 2, 3, 1],
+      "xaxis": "x2",
+      "yaxis": "y2"
+    }
+  ],
+  "layout": {
+    "grid": {"rows": 2, "columns": 1, "pattern": "independent"},
+    "xaxis": {
+      "side": "top",
+      "title": "MULTI-CATEGORY",
+      "range": [-0.5, 7],
+      "showline": true,
+      "tickfont": {"size": 16},
+      "ticks": "inside",
+      "ticklen": 10,
```

Comment:
Ah good eye, I didn't know we allowed negative `ticklen`. We probably shouldn't.

Reply:
Could someone applies a negative `ticklen` to reverse `ticks: 'outside'/'inside'` ?

Reply:
Ah

https://github.com/plotly/plotly.js/blob/e1ac60b1078b728277f2ba887cf5ad98b050f77b/src/plots/cartesian/layout_attributes.js#L348-L355

`ticklen` has `min: 0`, so that shouldn't work.

Reply:
Thanks for the reference.